File: 1.1/src/cz/mg/vulkan/PFNvkGetSemaphoreFdKHR.java
package cz.mg.vulkan;

public class PFNvkGetSemaphoreFdKHR extends VkFunctionPointer {
    public PFNvkGetSemaphoreFdKHR() {
    }

    protected PFNvkGetSemaphoreFdKHR(VkMemory vkmemory) {
        super(vkmemory);
    }

    protected PFNvkGetSemaphoreFdKHR(VkMemory vkmemory, long vkaddress) {
        super(vkmemory, vkaddress);
    }



    public PFNvkGetSemaphoreFdKHR(long value) {
        setValue(value);
    }

    public PFNvkGetSemaphoreFdKHR(VkInstance instance) {
        super(instance, new VkString("vkGetSemaphoreFdKHR"));
    }

    public int call(VkDevice device, VkSemaphoreGetFdInfoKHR pGetFdInfo, VkInt pFd){
        return callNative(getValue(), device != null ? device.getVkAddress() : VkPointer.getNullAddressNative(), pGetFdInfo != null ? pGetFdInfo.getVkAddress() : VkPointer.NULL, pFd != null ? pFd.getVkAddress() : VkPointer.NULL);
    }

    protected static native int callNative(long vkaddress, long device, long pGetFdInfo, long pFd);

}

File: 1.1/src/c/cz/mg/vulkan/PFN_vkGetSemaphoreFdKHR.c
#include <jni.h>
#include <stdlib.h>
#include <string.h>
#include <vulkan/vulkan.h>

jlong jniPointerToLong(const void* p);
void* jniLongToPointer(jlong l);
jlong jniFunctionPointerToLong(PFN_vkVoidFunction p);
PFN_vkVoidFunction jniLongToFunctionPointer(jlong l);
void jniThrowException(JNIEnv* env, const char* message);

jint Java_cz_mg_vulkan_PFNvkGetSemaphoreFdKHR_callNative(JNIEnv* env, jclass jc, jlong address, jlong device, jlong pGetFdInfo, jlong pFd){
    (void)env;
    (void)jc;
    PFN_vkGetSemaphoreFdKHR f = (PFN_vkGetSemaphoreFdKHR)jniLongToFunctionPointer(address);
    return f(
        *((VkDevice*)jniLongToPointer(device)),
        ((VkSemaphoreGetFdInfoKHR*)jniLongToPointer(pGetFdInfo)),
        ((int*)jniLongToPointer(pFd))
    );
}

